db/db_impl_write.cc

```diff
@@ -534,23 +566,76 @@ Status DBImpl::PipelinedWriteImpl(const WriteOptions& write_options,
   return w.FinalStatus();
 }
 
+Status DBImpl::UnorderedWriteMemtable(const WriteOptions& write_options,
+                                      WriteBatch* my_batch,
+                                      WriteCallback* callback, uint64_t log_ref,
+                                      SequenceNumber seq) {
+  PERF_TIMER_GUARD(write_pre_and_post_process_time);
+  StopWatch write_sw(env_, immutable_db_options_.statistics.get(), DB_WRITE);
+
+  WriteThread::Writer w(write_options, my_batch, callback, log_ref,
+                        false /*disable_memtable*/);
+
+  size_t total_count = 0;
+  size_t total_byte_size = 0;
+
+  if (w.CheckCallback(this)) {
+    if (w.ShouldWriteToMemtable()) {
+      w.sequence = seq;
+      total_count += WriteBatchInternal::Count(my_batch);
+    }
+    total_byte_size = WriteBatchInternal::AppendedByteSize(
+        total_byte_size, WriteBatchInternal::ByteSize(w.batch));
+  }
+
+  auto stats = default_cf_internal_stats_;
```

Comment:
`auto` here doesn't help readability. I suggest we spell out the type.

Reply:
I do agree with you on this point in general but in this particular case I believe this is a correct usage of auto. `stats` simply provides a shorter alias for `default_cf_internal_stats_`; if we had not used this alias, there would be no indication in the code that spells out the type of `default_cf_internal_stats_`. So auto doesn't decrease readability. But if you still feel strongly about it let me know.

Reply:
The criteria is not whether it decreases readability but whether it increases it. "auto is permitted when it increases readability".

Reply:
I believe this interpretation is not correct. But this is not worth arguing. I will go ahead with the suggested change.